task-sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -363,8 +363,6 @@ def xcom_pull(
             else:
                 xcoms.append(value)
 
-        if len(xcoms) == 1:
-            return xcoms[0]
```

Comment:
This would mean, though, that even if a single  XCom is requested directly with `task_ids="task_id"`, we would send a list.

Reply:
The behaviour we want (and test should be around it) is:

- A single value is returned when `task_ids="task1"`
- A list when `task_ids=["task1"]`

Reply:
There is also the case where it can be a single task but that but someone has requested multiple `map_indexes`

based on https://github.com/apache/airflow/blob/2.10.5/airflow/models/taskinstance.py#L625-L633

Reply:
Thanks, i have handled it